prow/pod-utils/gcs/upload.go

```diff
@@ -150,47 +152,73 @@ func FileUploadWithOptions(file string, opts pkgio.WriterOptions) UploadFunc {
 			}
 		}
 
-		uploadErr := DataUploadWithOptions(reader, opts)(writer)
+		firstTimeNewReader := true
+		newReader := func() (io.ReadCloser, error) {
+			// The first time the reader is opened already as we needed it to
+			// gather some info and tune the BufferSize.
+			if firstTimeNewReader {
```

Comment:
tbh I'm not entirely sure that `firstTimeNewReader` serves a purpose, as no matter first time or second time, the logic is pretty much identical:
- `os.Open(file)`
- upload with the same `opts`

am I missing something?

Reply:
I'm going to simplify it as you described and omit this (pointless?) micro-optimization.